plan/plan.go

```diff
@@ -164,8 +166,14 @@ type PhysicalPlan interface {
 	json.Marshaler
 	Plan
 
-	// matchProperty means that this physical plan will try to return the best plan that matches the required property.
-	// childPlanInfo means the plan infos returned by children.
+	// matchProperty calculates the cost of the physical plan if it matches the required property.
+	// It's usually called at the end of convert2PhysicalPlan, some PhysicalPlan does not implement it as there is
+	// no property to match, they just do the cost calculation directly.
+	// If not match, the cost will be set to MaxInt64, so it will not be chosen as the result physical plan.
```

Comment:
If the cost of the physical plan does not match the required property, the cost will be set to MaxInt64 so it will not be chosen as the result physical plan.